tests/cnf/core/network/sriov/tests/rdmametrics.go

```diff
@@ -0,0 +1,368 @@
+		} `json:"device-info,omitempty"`
+	}
+
+	var annotation []PodNetworkStatusAnnotation
+
+	err := json.Unmarshal([]byte(tPod.Object.Annotations["k8s.v1.cni.cncf.io/network-status"]), &annotation)
+	Expect(err).ToNot(HaveOccurred(), "Failed to Unmarshal Pod Network Status annotation")
+
+	for _, annotationObject := range annotation {
+		if annotationObject.Interface == podInterface {
+			if annotationObject.DeviceInfo.Pci.PciAddress != "" || annotationObject.DeviceInfo.Pci.RdmaDevice != "" {
+				return annotationObject.DeviceInfo.Pci.PciAddress, annotationObject.DeviceInfo.Pci.RdmaDevice, nil
+			}
+
+			return "", "", fmt.Errorf("PCI Address or Rdma Device are not present in the Interface %v", annotationObject)
+		}
+	}
+
+	return "", "", fmt.Errorf("interface %s not found", podInterface)
+}
+
+func defineAndCreateNodePolicy(polName, resName, sriovInterface string, numVfs, endVf int) *sriov.PolicyBuilder {
+	testPolicy := sriov.NewPolicyBuilder(APIClient,
+		polName, NetConfig.SriovOperatorNamespace, resName, numVfs, []string{sriovInterface}, NetConfig.WorkerLabelMap).
+		WithDevType("netdevice").
+		WithVFRange(0, endVf).
+		WithRDMA(true)
+	err := sriovenv.CreateSriovPolicyAndWaitUntilItsApplied(testPolicy, tsparams.MCOWaitTimeout)
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to create SriovNodePolicy %s", polName))
+
+	return testPolicy
+}
+
+func defineAndCreateSriovNetworkWithRdma(netName, resName string, withRdma bool) *sriov.NetworkBuilder {
+	testNetBuilder := sriov.NewNetworkBuilder(
+		APIClient, netName, NetConfig.SriovOperatorNamespace, tsparams.TestNamespaceName, resName).
+		WithMacAddressSupport().
+		WithIPAddressSupport().
+		WithStaticIpam().
+		WithLogLevel(netparam.LogLevelDebug)
+
+	if withRdma {
+		testNetBuilder.WithMetaPluginRdma()
+	}
+
+	testNetwork, err := testNetBuilder.Create()
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to create Sriov Network %s", netName))
+
+	return testNetwork
+}
+
+func defineAndCreatePod(netName1, netName2 string) *pod.Builder {
+	var netAnnotation []*types.NetworkSelectionElement
+	if netName2 == "" && netName1 != "" {
+		netAnnotation = []*types.NetworkSelectionElement{
+			{
+				Name:       netName1,
+				MacRequest: tsparams.ServerMacAddress,
+				IPRequest:  []string{tsparams.ServerIPv4IPAddress},
+			},
+		}
+	} else {
+		netAnnotation = []*types.NetworkSelectionElement{
+			{
+				Name:       netName1,
+				MacRequest: tsparams.ServerMacAddress,
+				IPRequest:  []string{tsparams.ServerIPv4IPAddress},
+			},
+			{
+				Name:       netName2,
+				MacRequest: tsparams.ClientMacAddress,
+				IPRequest:  []string{tsparams.ClientIPv4IPAddress},
+			},
+		}
+	}
+
+	tPod, err := pod.NewBuilder(APIClient, "testpod", tsparams.TestNamespaceName, NetConfig.CnfNetTestContainer).
+		WithSecondaryNetwork(netAnnotation).
+		WithPrivilegedFlag().
+		CreateAndWaitUntilRunning(2 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create Pod %s")
+
+	tPod.Exists()
```

Comment:
?

Reply:
`CreateAndWaitUntilRunning()` returns the pod builder which is still in Pending status. So it does not have network-status annotations set. `Exists()` updates the builder with all fields, so we can see the annotations.

Ideally this should be part of `CreateAndWaitUntilRunning` method.